Rename exported JavaScript modules to include component name (#2426)
alphagov/govuk-frontend: CHANGELOG.md

This change was introduced in [pull request #2417: Remove deprecated `govuk-tag--inactive class`](https://github.com/alphagov/govuk-frontend/pull/2417).

> #### Make sure any individually imported JavaScript modules work as expected

Reviewer: ```suggestion
#### Make sure individually imported JavaScript modules work as expected
```

> #### Component JavaScript modules have a unique name when imported individually

Reviewer 2: I _think_ we've sort of settled on using a description of 'what the user needs to do' for breaking changes (for example 'Make sure character counts still work as expected') – is there an equivalent we could use for this?

I also think we might want to list this under 'Fixes' as well, as it feels like a bug?

Author: @EoinShaughnessy I've had a go at the release notes for this (including Ollie's suggestion here), but I think they could do with some love! Would it be helpful if I put some time in to talk through it together?

Reviewer: @vanitabarrett That'd be great, thanks!

Author: @EoinShaughnessy Let's have a chat about it at our catch-up tomorrow 👍

> We've changed the naming of our component JavaScript modules so that individual imports are now attached to

Author: Yeah, I found it a bit difficult to find the right wording here! At the moment, I can't think of alternative though 🤔

`window.GOVUKFrontend.[ComponentName]` instead of `window.GOVUKFrontend`.

> This means you can now import multiple modules without overwriting the previous one, for example:

Reviewer: ```suggestion
You can now import multiple modules without overwriting the previous one, for example:
```

```
//= require govuk/components/accordion/accordion.js
//= require govuk/components/button/button.js

> # The above are available under window.GOVUKFrontend.Accordion and window.GOVUKFrontend.Button respectively

Reviewer: ```suggestion
# These modules are available under window.GOVUKFrontend.Accordion and window.GOVUKFrontend.Button respectively
```

```

> If you are importing indvidual component JavaScript, you should check any references to `window.GOVUKFrontend` in your code and update them to point to the correct component: `window.GOVUKFrontend.[ComponentName]`. You may have implemented workarounds that can now be removed.

Reviewer: ```suggestion
If you're importing individual component JavaScript, you should check any references to `window.GOVUKFrontend` in your code and update them to point to the correct component: `window.GOVUKFrontend.[ComponentName]`. You can now remove any workarounds you may have implemented.
```
Wonder if we could change "If you're importing individual component JavaScript" to "If you're importing JavaScript modules individually" (to somewhat mirror wording in heading title)? 
Also, should we tell users "you can" or "you should" now remove any workarounds?

> This change does not affect you if you followed our [Getting Started guide](https://frontend.design-system.service.gov.uk/get-started/#5-get-the-javascript-working) and are importing all of the GOV.UK Frontend JavaScript in one go via `all.js`, or if you installed GOV.UK Frontend using precompiled files.

Reviewer: Wonder should this display right under the heading? Just to help users who do not need to do anything skip over this content.

```suggestion
You do not need to do anything if you have either:

- followed our [Getting Started guide](https://frontend.design-system.service.gov.uk/get-started/#5-get-the-javascript-working) and are importing all of the GOV.UK Frontend JavaScript in one go via `all.js`
- installed GOV.UK Frontend using precompiled files

```

Author: Good idea, I've moved this to the top and made the suggestions you mentioned

> This change was introduced in [pull request #1836: Rename exported JavaScript modules to include component name](https://github.com/alphagov/govuk-frontend/issues/1836)]

Reviewer: ```suggestion
This change was introduced in [pull request #1836: Rename exported JavaScript modules to include component name](https://github.com/alphagov/govuk-frontend/issues/1836)].
```
